src/widget/woverview.cpp

```diff
@@ -313,11 +330,24 @@ void WOverview::receiveCuesUpdated() {
 }
 
 void WOverview::mouseMoveEvent(QMouseEvent* e) {
-    if (m_orientation == Qt::Horizontal) {
-        m_iPos = math_clamp(e->x(), 0, width() - 1);
-    } else {
-        m_iPos = math_clamp(e->y(), 0, height() - 1);
+    if (m_bDrag) {
+        if (m_orientation == Qt::Horizontal) {
+            m_iPos = math_clamp(e->x(), 0, width() - 1);
+        } else {
+            m_iPos = math_clamp(e->y(), 0, height() - 1);
+        }
+    }
+
+    for (auto& mark : m_marks) {
+        WaveformMarkProperties markProperties = mark->getProperties();
+        if (markProperties.m_renderedArea.contains(e->pos())) {
+            markProperties.m_bMouseHovering = true;
```

Comment:
If we set here only one to the hovering state the loop in painting can probably simplified.

We can make use of a timer here, to make this feature less attracting. It is actually not a simple hover effect, it is a kind of tooltip. So I think it should come and go with a delay.

I think also if a user wants to read the whole lable, all other lales are irrelevant and can be hidden. This will hopefully declutter the view even more.

Later we an considder to show even more Infos for a cue in this state.

Reply:
You're right, it makes sense to only set m_bMouseHovering for one mark. I moved the logic from drawMarks to mouseMoveEvent.

I think a timer as well as hiding all labels (not just those with text that would overlap) would be superfluous. This works intuitively already.

> Later we an considder to show even more Infos for a cue in this state.

Yes, showing the time position and the time until reaching the cue would be useful.

Reply:
> I think a timer as well as hiding all labels (not just those with text that would overlap) would be superfluous. This works intuitively already.

Ok, right, since the hover region is relative small, there is only a minor chance to get a distracting flickering when using the mouse.